package/MDAnalysis/analysis/msd.py

```diff
@@ -421,3 +456,46 @@ def _conclude_fft(self):  # with FFT, np.float64 bit prescision required.
                 positions[:, n, :]
             )
         self.results.timeseries = self.results.msds_by_particle.mean(axis=1)
+    
+
+    def _conclude_modified(self):
+        from collections import defaultdict
+        print("The Dump file has non-linear time spacing")
+        
+        dump_times = [ts.time for ts in self._trajectory]
```

Comment:
This iterates through the whole trajectory and may already be expensive!

Maybe unavoidable but keep it in mind.